source/fab/tools/psyclone.py

```diff
@@ -24,45 +26,163 @@ class Psyclone(Tool):
     '''This is the base class for `PSyclone`.
     '''
 
-    def __init__(self, api: Optional[str] = None):
+    def __init__(self):
         super().__init__("psyclone", "psyclone", Category.PSYCLONE)
-        self._api = api
+        self._version = None
+
+    def check_available(self) -> bool:
```

Comment:
Can this whole complication, or a good section of it, be replaced by making use of `shutil.which(...)`?

Reply:
Interesting, I didn't know that tool (I know the shell tool, but afaik that's not available everywhere).
I still need to actually run `psyclone`, since e.g. on NCI we get:
```
$ psyclone
/apps/python3/3.12.1/bin/python3.12: error while loading shared libraries: libpython3.12.so.1.0: cannot open shared object file: No such file or directory
$ which psyclone
~/.local/bin/psyclone
$ python3
>>> import shutil
>>> shutil.which("psyclone")
'/home/903/jxh903/.local/bin/psyclone'
```

So, while indeed this is a kind of broken installation, I can't rely on `shutil.which` alone. But I'll check if this simplifies the code further.

Reply:
Plus, I also still need the version number to switch the command line options between 2.5 and 3.0. But I also think that code is now a lot simpler :)

Reply:
I've raised a ticket #393 to investigate this issue further. In the mean time this approach is reasonably neat.